civihr_default_theme/template.php

```diff
@@ -660,61 +660,97 @@ function _hide_menu_items(&$element) {
-    }
 
-    $element['#localized_options']['attributes']['data-target'] = '#';
-  }
+  _add_active_class_if_link_is_current($element);
+  _add_unique_class_to_link($element);
+  _hide_admin_link_to_basic_users($element);
+
+  return '<li' . drupal_attributes($element['#attributes']) . '>' . _get_link_markup($element) . "</li>\n";
+}
 
-  // Fix for active class.
-  if (($element['#href'] == current_path() || ($element['#href'] == '<front>' && drupal_is_front_page())) && (empty($element['#localized_options']['language']) || $element['#localized_options']['language']->language == $language_url->language)) {
-    $element['#attributes']['class'][] = 'active';
+/**
+ * If the link is for the current page, then it adds the active class to it
+ *
+ * @param array $link
+ */
+function _add_active_class_if_link_is_current(&$link) {
+  if (($link['#href'] == current_path() || ($link['#href'] == '<front>' && drupal_is_front_page())) && (empty($link['#localized_options']['language']) || $link['#localized_options']['language']->language == $language_url->language)) {
+    $link['#attributes']['class'][] = 'active';
   }
 
   // Add active class to li if active trail.
-  if (in_array('active-trail', $element['#attributes']['class'])) {
-    $element['#attributes']['class'][] = 'active';
+  if (in_array('active-trail', $link['#attributes']['class'])) {
+    $link['#attributes']['class'][] = 'active';
   }
+}
 
-  // Add a unique class using the title.
-  $title = strip_tags($element['#title']);
-  $element['#attributes']['class'][] = 'menu-link-' . drupal_html_class($title);
+/**
+ * Adds a unique class to a link using its title
+ *
+ * @param array $link
+ */
+function _add_unique_class_to_link(&$link) {
+  $title = strip_tags($link['#title']);
+  $link['#attributes']['class'][] = 'menu-link-' . drupal_html_class($title);
+}
 
-  /* Code Added */
-  _hide_menu_items($element);
-  /* End - Code Added */
-  if ($element['#title'] === 'Manager Leave') {
-    $element['#localized_options']['html'] = true;
-    $output = l(_get_pending_leave_request_markup($element), $element['#href'], $element['#localized_options']);
-  } else {
-    $output = l($element['#title'], $element['#href'], $element['#localized_options']);
+/**
+ * Hide the link to the admin if the current user does not
+ * have administer access
+ *
+ * @param array $link
+ */
+function _hide_admin_link_to_basic_users(&$link) {
+  $adminAccess = user_access("administer CiviCRM");
+  $identifier = $link['#localized_options']['identifier'];
+
+  if ($identifier == 'main-menu_civihr-admin:civicrm' && !$adminAccess) {
+    $link['#attributes']['class'][] = 'hidden';
   }
+}
 
-  return '<li' . drupal_attributes($element['#attributes']) . '>' . $output . $sub_menu . "</li>\n";
+/**
+ * Build and returns the markup of the given link (and submenu, if present)
+ *
+ * @param array $link
+ *
+ * @return string
+ */
+function _get_link_markup(&$link) {
```

Comment:
It looks like this function only works for menu links, so I think it would be better to give it a more specific name, like `_get_link_markup()`